src/dependency/dependencyManager.ts

```diff
@@ -99,7 +99,15 @@ function getPackagesForCurrentPlatform(packages: AbsolutePathPackage[], platform
 function filterInstalledPackages(packages: AbsolutePathPackage[]) {
     if (packages) {
         return packages.filter(pkg => {
-            return !installFileExists(pkg.installPath, InstallFileType.Finish);
+            if (!installFileExists(pkg.installPath, InstallFileType.Finish)) {
+                return true;
+            }
+            let installedPackageIntegrity = fs.readFileSync(getInstallLockFilePath(pkg.installPath, InstallFileType.Finish)).toString().trim();
```

Comment:
should this be called outside the filter? now it has been called mutiple times.
And then the filter can be simplified as well

Reply:
it should be applied for each package, so it better to be kept inside the filter